test/test_autograd.py

```diff
@@ -12926,6 +12926,409 @@ def run_test(fn):
         run_test(lambda x: x.transpose_(0, 1))
 
 
+NUM_GPU_CYCLES_IN_ONE_SEC = 2_000_000_000
+
+
+class TestAutogradStreamSynchronization(TestCase):
+    def get_default_streams(self):
+        with torch.cuda.device(0):
+            default_stream_0 = torch.cuda.default_stream()
+        with torch.cuda.device(1):
+            default_stream_1 = torch.cuda.default_stream()
+        return default_stream_0, default_stream_1
+
+    def synchronize_all_devices(self):
+        torch.cuda.synchronize(0)
+        torch.cuda.synchronize(1)
+
+    def assert_all_streams_default(self):
+        # Sanity check
+        d0, d1 = self.get_default_streams()
+        self.assertEqual(torch.cuda.current_stream(0), d0)
+        self.assertEqual(torch.cuda.current_stream(1), d1)
+
+    @unittest.skipIf(not TEST_CUDA, "requires CUDA")
```

Comment:
```suggestion
    @unittest.skipIf(not TEST_CUDA, "requires CUDA")
    @unittest.skipIf(torch.cuda.device_count() < 2, "GPU count is less than 2")
```

Reply:
Whoops, I should only require a single device in this test, but my helper functions are hardcoded to reference two